core/app/models/spree/promotion_code.rb

```diff
@@ -39,4 +40,8 @@ def usage_limit
   def downcase_value
     self.value = value.downcase
   end
+
+  def remove_excess_whitespace
+    value&.gsub! /\s+/, ''
```

Comment:
I assume the "ambiguous first argument" is indicating I should add parens. So `gsub!(/\s+/, '')` (under the idea that somehow that is clearer).

Is the `unexpected token error` referring to the safe navigation operator? Is this a project decision (compat with older Ruby) or just hound not knowing the more modern syntax?